nemo_skills/pipeline/train.py

```diff
@@ -31,6 +32,135 @@
 class TrainingAlgo(str, Enum):
     sft = "sft"
     dpo = "dpo"
+    rm = "rm"
+
+
+@dataclass
+class TrainingParams:
+    config_name: str
+    config_path: str
+    nemo_model: str
+    output_dir: str
+    training_data: str
+    validation_data: str
+    num_gpus: int
+    num_nodes: int
+    expname: str
+    training_algo: str
+    disable_wandb: bool
+    wandb_project: str
+    timeout: str
+    extra_arguments: str = ""
+    logging_params: str = ""
+
+
+def train_sft(params: TrainingParams) -> str:
+    if params.config_name is None:
+        config_name = "sft_config"
+    else:
+        config_name = params.config
+    config_params = f"--config-name={config_name} --config-path={params.config_path} "
```

Comment:
Since we are not sharing the main cmd, why not include these 3 arguments directly in the command below instead of creating them here and then formatting inside cmd?

Reply:
Alternatively, if that cmd is fully shared across all 3 cases I think it's ok to move it inside a separate function and reuse that. It's less explicit but still close enough in the code that hopefully won't confuse anyone. Whichever you think makes more sense

Reply:
Fair enough--yeah I refactored out the shared part of the script and threw all of the customization into the top layer command.